### PR TITLE
Feature sshreceiver

### PR DESCRIPTION
**Pull request by nslaughter**
## tldr

I would like your input on ...
(a) should we tweak the code to align more directly with other popular SSH checks 
(b) should we add a config option to ignore known_hosts and
(c) general architecture, quality suggestions?

In consideration of your time, please don't dive too deeply into whether the tests in `internal/configssh` are passing at the moment. There are relatively minor changes occurring there to integrate changes in how we handle known_hosts. On the other hand, very interested in feedback on whether you believe the specified cases cover the right things.

## whether the metrics compute what you'd like/expect

**_The way this code works:_** 
If SFTPEnabled is not set then the resulting metrics are success and duration, which is measured as the time to connect.
If SFTP Enabled is set then we also send a read request then provide success based on whether the request succeeded along with duration.

**_The differences in how other SSH checks works:_**
In another popular SSH check implementation, I have seen the following...
1) SFTP is enabled by default.
2) While SSH reports connection (success) without SFTP, there is no duration metric for SSH alone.

## handling key not in known_hosts

Another popular SSH check provides an option for adding keys to known_hosts automatically. That's the last step of implementation, which is in process now.

## the usual feedback

Also welcome any feedback on usual aspects of code review: clarity, quality, tests, architecture, etc

### CODE REVIEW

File: receiver/sshcheckreceiver/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+| ------------------------ |-----------|
+| Stability                | [beta]    |
+| Supported pipeline types | metrics   |
+| Distributions            | [contrib] |
+
+This receiver fetches stats from a ssh server using the `/_node/{node-name}/_stats/ssh` [endpoint](https://docs.ssh.org/en/latest/api/server/common.html#node-node-name-stats).
+
+## Prerequisites
+
+By default this receiver requires the server key in known_hosts.
```
Comment (carlosalberto) [inline receiver/sshcheckreceiver/README.md:13]: Other similar checks do not _seem_ to require this. Any idea about this? Or is it that those other ones don't document this?

File: receiver/sshcheckreceiver/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## Configuration
+
+The following settings are required:
+- `username`
+- `password`
+
+The following settings are optional:
+- `endpoint` (default: `localhost:2222`): The address of the SSH endpoint
```
Comment (carlosalberto) [inline receiver/sshcheckreceiver/README.md:22]: There's no uniformity regarding default or no default on this in the Collector receivers, BUT I'd vow to make it required (we can relax it later if/as needed).

File: receiver/sshcheckreceiver/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+- `collection_interval` (default = `60s`): This receiver collects metrics on an interval. This value must be a string readable by Go's [time.ParseDuration](https://pkg.go.dev/time#ParseDuration). Valid time units are `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.
```
Comment (carlosalberto) [inline receiver/sshcheckreceiver/README.md:24]: Similarly, some receivers do mention the Go's parseability bit, but I'd vow to remove it, i.e. "This receiver collects metrics on an interval. Valid time units are `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`." should be enough IMHO

File: receiver/sshcheckreceiver/documentation.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+## Metrics
+
+These are the metrics available for this scraper.
+
+| Name | Description | Unit | Type | Attributes |
+| ---- | ----------- | ---- | ---- | ---------- |
+| **sshcheck.duration** | Measures the duration of SSH check. | ms | Sum(Int) | <ul> <li>ssh.endpoint</li> </ul> |
+| sshcheck.sftp_duration | Measures SFTP response duration. | ms | Sum(Int) | <ul> <li>ssh.endpoint</li> </ul> |
+| **sshcheck.sftp_success** | The SFTP request succeeded. | 1 | Gauge(Int) | <ul> <li>ssh.endpoint</li> </ul> |
```
Comment (carlosalberto) [inline receiver/sshcheckreceiver/documentation.md:13]: Any reason why this is reported by default too? `sshcheck.sftp_duration` is not. Mostly curiosity ;)

File: receiver/sshcheckreceiver/scraper.go [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+func (s *sshcheckScraper) start(ctx context.Context, host component.Host) (err error) {
+	s.SSHClient, err = s.Config.ToClient()
+	return
+}
+
+// scrape connects to the endpoint and produces metrics based on the response
+func (s *sshcheckScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
+	now := pcommon.NewTimestampFromTime(time.Now())
+
+	if s.SSHClient == nil {
```
Comment (carlosalberto) [inline receiver/sshcheckreceiver/scraper.go:49]: Move this up at the start of the method? Before creating the timestamp, that is.

File: receiver/sshcheckreceiver/README.md (hunk 1 above)
Comment (nslaughter) [reply receiver/sshcheckreceiver/README.md:13]: We can provide an option to automatically add an entry in known_hosts. Unlike what I thought previously, I don't believe we need to provide a setting for the path of a file. They are referenced by path in docs for `man sshd`.

File: receiver/sshcheckreceiver/README.md (hunk 2 above)
Comment (nslaughter) [reply receiver/sshcheckreceiver/README.md:22]: Makes sense. To support the notion... users will never use a default for this.

File: receiver/sshcheckreceiver/README.md (hunk 3 above)
Comment (nslaughter) [reply receiver/sshcheckreceiver/README.md]: changed

File: receiver/sshcheckreceiver/README.md (hunk 2 above)
Comment (nslaughter) [reply receiver/sshcheckreceiver/README.md]: changed.

File: receiver/sshcheckreceiver/README.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
+# CouchDB Receiver
+
+| Status                   |           |
+| ------------------------ |-----------|
+| Stability                | [beta]    |
+| Supported pipeline types | metrics   |
+| Distributions            | [contrib] |
+
+This receiver fetches stats from a ssh server using the `/_node/{node-name}/_stats/ssh` [endpoint](https://docs.ssh.org/en/latest/api/server/common.html#node-node-name-stats).
```
[the remark this answers is not in the document]
Comment (nslaughter) [reply receiver/sshcheckreceiver/README.md]: oops

File: receiver/sshcheckreceiver/README.md (hunk 1 above)
Comment (nslaughter) [reply receiver/sshcheckreceiver/README.md]: The way this works now is that an `.IgnoreHostKey` setting is provided to allow insecure connections. If this is not set then we use known_hosts files to validate server/key.
